app/client/cli/servicer.go

```diff
@@ -0,0 +1,284 @@
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/pocket/shared/crypto"
+)
+
+func init() {
+	rootCmd.AddCommand(NewServicerCommand())
+}
+
+// TECHDEBT: (unittest) unit test the command: e.g. on number of arguments
+func NewServicerCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "Servicer",
+		Short:   "Servicer specific commands",
+		Aliases: []string{"servicer"},
+		Args:    cobra.ExactArgs(0),
+	}
+
+	cmds := servicerCommands()
+	applySubcommandOptions(cmds, attachPwdFlagToSubcommands())
+	applySubcommandOptions(cmds, attachKeybaseFlagsToSubcommands())
+
+	cmd.AddCommand(cmds...)
+
+	return cmd
+}
+
+func servicerCommands() []*cobra.Command {
+	cmdDef := actorCmdDef{"Servicer", coreTypes.ActorType_ACTOR_TYPE_SERVICER}
+	cmds := []*cobra.Command{
+		newStakeCmd(cmdDef),
+		newEditStakeCmd(cmdDef),
+		newUnstakeCmd(cmdDef),
+		newUnpauseCmd(cmdDef),
+		{
+			Use:   "Relay <servicer> <application> <relayChainID> <payload>",
+			Short: "Relay <servicer> <application> <relayChainID> <payload>",
+			Long: `Sends a trustless relay using <payload> as contents, to the specified active <servicer> in the the <application>'s session.
+Will prompt the user for the *application* account passphrase`,
+			Aliases: []string{},
+			Args:    cobra.ExactArgs(4),
+			RunE: func(cmd *cobra.Command, args []string) error {
+				servicerAddr := args[0]
+				applicationAddr := args[1]
+				chain := args[2]
+				relayPayload := args[3]
+
+				// TODO: (SUGGESTION) refactor to decouple the client logic from the CLI/command
+				pk, err := getPrivateKey(applicationAddr)
+				if err != nil {
+					return fmt.Errorf("error getting application's private key: %w", err)
+				}
+
+				session, servicer, err := fetchServicer(cmd.Context(), applicationAddr, chain, servicerAddr)
+				if err != nil {
+					return fmt.Errorf("error getting servicer for the relay: %w", err)
+				}
+
+				relay, err := buildRelay(relayPayload, pk, session, servicer)
+				if err != nil {
+					return fmt.Errorf("error building relay from payload: %w", err)
+				}
+
+				fmt.Printf("sending trustless relay for %s to %v with payload: %s\n", applicationAddr, servicer, relayPayload)
+
+				resp, err := sendTrustlessRelay(cmd.Context(), servicer.ServiceUrl, relay)
+				if err != nil {
+					return err
+				}
+
+				fmt.Printf("HTTP status code: %d\n", resp.HTTPResponse.StatusCode)
+				fmt.Println("Response: ", resp.JSON200)
+
+				return nil
+			},
+		},
+	}
+
+	return cmds
+}
+
+// TODO: (QUESTION): do we need/want a cli subcommand for fetching servicers?
+
+// fetchServicer returns the servicer specified by the <servicer> argument.
+// It validates the following conditions:
+//
+//	A. The <application> argument is the address of an active application
+//	B. The <servicer> is the address of a servicer that is active in the application's current session.
+//
+// TODO: (SUGGESTION) use a package-internal interface for servicer and application?
```

Comment:
We used to have many package specific interfaces for every type (e.g. the Validator interface in consensus, the Validator interface in persistence, etc) and had 6 different Validator interfaces at one point.

I'm open to either approach but would push back against this just because it got really confusing in the past.

Reply:
Thanks, makes sense. This was a TODO from the first time I read the code. Doesn't look necessary on a second/third pass.